### DISCUSSION TITLE
[POLL] Our policy about quality of players and ability to play streams by them

### DISCUSSION THREAD

**Discussion by Dum4G** [iptv-org/community#561] [Ideas]
Historically, we reffered to VLC as a sweet spot for stream verification as it was capable of handling adaptive streams in most formats as well as user agent and refferer spoofing. It was good enough to ensure the stream will be available for most viewers. However, during last year some sources have modernized the way they treat browser requests. VLC has started to be detected and fallen short against actual browsers and many other players out there that handle adaptive links in more appropiate ways. This situation leads us to a question: are we going to stick to VLC or move forward and advise something better?

If we move forward:
`+` It would be possible to approve some more publically available streams
`+` Users of "advanced" players would be capable to watch more channels from the repo
`+` Perhaps, it would be possible to add some more non-encrypted .mpd sources like cliptv.az
`-` We'd have to redefine our requirements for a "sweet spot" player
`-` We'd have to retest and reevaluate every solution that is currently mentioned in awesome-iptv repo. Some of them might be either excluded  or rated as less worthy
`-` We'd have to encourage people to move to better players and sometimes to even switch for a proper device running proper OS for a proper player. That means we'd have to change our main readme file and discourage people from using VLC or less worthy solutions from now on.
`-` It's possible that we would have to deal with more reports of broken channels coming from people that used to use VLC-based players with VLCs native adaptive link impl or even more primitive solutions

If we keep the same approach:
`+` The playlist content will remain reachable for all viewers no matter what kind of solution or device they use
`+` No additioinal work for maintainers
`+` In theory it would be possible to automate stream verification using VLC as a command-line tool
`-` Some channels that might be interesting for the audience won't be present in playlist
`-` Users of "advanced" players would have to fork the playlist and add some sources on their own or switch to alternatives
`-` Stream sources that are not playing in VLC will be rejected no matter of it's public availability or lack of any other restrictions

Vote and share your thoughts

**Comment by AntiPontifex** [comment]
If I remember correctly, Dum4G, if you go to VLCs website and look under their upcoming "Google Summer of Code" item, they are planning to add mpd support. In the meantime, developing a consensus "go to" player would be advantageous. My .02 anyway.
